pkg/sysfs/system.go

```diff
@@ -995,3 +1017,40 @@ func (sys *system) discoverCache(path string) error {
 
 	return nil
 }
+
+var eppStrings = func() [EPPUnknown]string {
+	var e [EPPUnknown]string
+	e[EPPPerformance] = "performance"
+	e[EPPBalancePerformance] = "balance_performance"
+	e[EPPBalancePower] = "balance_power"
+	e[EPPPower] = "power"
+	return e
+}()
+
+// String returns EPP value as string
+func (e EPP) String() string {
+	if int(e) < len(eppStrings) {
+		return eppStrings[e]
+	}
+	return ""
+	/*switch e {
+	case EPPPerformance:
+		return "performance"
+	case EPPBalancePerformance:
+		return "balance_performance"
+	case EPPBalancePower:
+		return "balance_power"
+	case EPPPower:
+		return "power"
+	}*/
+}
+
+// EPPFromString converts string to EEP value
+func EPPFromString(s string) EPP {
+	for e, v := range eppStrings {
+		if s == v {
+			return EPP(e)
+		}
+	}
+	return EPPUnknown
+}
```

Comment:
nitpick: considering all the trouble we go through just to be able to use a fixed-sized array instead of a slice for Epp.String(), wouldn't it then be justified and more symmetric to use a map instead of iteration for the other direction, with something like this ? 
```
var namedEPPs = func() map[string]EPP {
    m := map[string]EPP{}
    for epp, str := range eppStrings {
        m[str] = epp
    }
    return m
}()

func EPPFromString(name string) EPP {
    if epp, ok := namedEPPs[name]; ok {
        return epp
    }
    return EPPUnknown
}
```

Reply:
Mm, not a nitpick. The code was quite hastily written and I wasn't totally happy with it myself. Your suggestion would be an improvement. Maybe this is a good place to discuss what would be the sanest/safest way to specify enum-string types like this(?) I don't have a good answer, maybe you do 😄 

I tried a bit different approach utilizing the stringer tool in the new version of the patch. That should better ensure that the enums and strings are kept in sync with each other. Not that I think there would really be any changes here, but, at least as an excercise...

Reply:
> I don't have a good answer, maybe you do 

Nope, unfortunately I don't. But having seen this updated patch set, I have to say that stringer really seems half-arsed to me: you still need to manually write the string-to-type direction/function, and that ends up being half of your original code without stringer. Plus a bit of extra boilerplate. And you have to remember to manually run `make generate` when you change something. Almost as if the cure would become the poison.
 
> I tried a bit different approach utilizing the stringer tool in the new version of the patch. That should better ensure that the enums and strings are kept in sync with each other. Not that I think there would really be any changes here, but, at least as an excercise...

I'm not opposing this, but I sure don't like it either. If others are fine with this, I'm fine with it, too.